pagopa/pn-frontend
# migration to bff api - pn-9831

## Short description
In this pr we migrate the notification detail api to that hosted by the bff

## List of changes proposed in this pull request
- Notification.api.ts (pa/pf/pg) removed the call to the parsing function
- notifications.routes.ts (pa/pf/pg) changed the path to the api
- Updated tests and mocks

## How to test
DEV environment
- grossini -> palermo -> GPDJ-UGUM-XHUR-202402-L-1 -> check that RADD disclaimer is shown
- for any user and any notification -> check that AAR box is correctly filled and that download works
- log with a pa -> cancel a notification -> check that the status is correctly changed to cancellation in progress
- log with a pf/pg -> find the notification cancelled in the previous step -> check that the status is correctly changed to cancellation
- for any user and any notification -> check that the timeline is ordered from the most recent event to the oldest one
- for any user and any notification -> check that the events under the ACCEPTED status are moved to the next one, leaving the legal facts unmoved
- grossini -> sappada -> TXKN-RWMX-KNAJ-202403-D-1 -> check that a AAR legal fact is added under the ANALOG_FAILURE_WORKFLOW event
- for any user and any notification -> check that the copy under DELIVERED status changes based on delivery mode (digital or analogical)
- for any user and any notification -> notification viewed by a delegate -> check that the copy under VIEWED status has informations about the delegate that viewed the notification
- grossini -> sappada -> RDUL-LZNL-QUPL-202403-D-1 -> check that DELIVERED status is empty and its events have been moved to the DELIVERING status
